extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -388,8 +388,17 @@ export class JupyterServerInstallation {
 	/**
 	 * Prompts user to upgrade certain python packages if they're below the minimum expected version.
 	 */
-	public promptForPackageUpgrade(): Promise<void> {
-		return this.upgradePythonPackages(true, false);
+	public async promptForPackageUpgrade(): Promise<void> {
+		if (this._installInProgress) {
+			return;
```

Comment:
Instead of just silently returning it might be useful to alert the user that some packages need to be updated but that they can't because another install is in progress and they should try again later. Possibly even have the dialog have a retry button so they can wait for the install to complete and then re-run this check?

Reply:
Should we have command palette task for package upgrades, or would that be too easy to confuse with the general install tasks? Currently we just silently downlevel packages in the background, so if we want to have a retry mechanism then a command palette task seems good.

CC @chlafreniere

Reply:
Yeah that seems like a pretty useful thing to have. Call it **Check for Package Updates** or something.